### ISSUE TITLE
Fix BMN model's structure

### ISSUE THREAD

**Issue by FuNian788** [JJBOY/BMN-Boundary-Matching-Network#33]
Hi, I am a graduate student who is interseted in TAD and hope more communication.
First, thank you for the implementation of BMN. I find BMN model's structure is not the same as introduced in the paper, so I checked the source code by Tianwei Lin. I think the output dimension of x_1d_p and input dimension of x_3d_p may be changed to 128(hidden_dim_2d) as follows:
`        # Proposal Evaluation Module
        self.x_1d_p = nn.Sequential(
            nn.Conv1d(self.hidden_dim_1d, self.hidden_dim_2d, kernel_size=3, padding=1),
            nn.ReLU(inplace=True)
        )
        self.x_3d_p = nn.Sequential(
            nn.Conv3d(self.hidden_dim_2d, self.hidden_dim_3d, kernel_size=(self.num_sample, 1, 1),stride=(self.num_sample, 1, 1)),
            nn.ReLU(inplace=True)
        )`
Lin's source code in this link:
https://github.com/PaddlePaddle/models/blob/release/1.8/PaddleCV/video/models/bmn/bmn_net.py#L141-L149

Best wishes!

**Comment by JJBOY** [comment]
Yes, you are right. But I think this will not affect the perfermance. And in my newest version, I also changed the BM map. In his paper and code,  the shape of BM map is D\*T, where D represents duration and T represents start time, but the shape is T\*T represnets for start time \* end time , for the alignment with the AAAI paper DBG.

**Comment by FuNian788** [comment]
Thanks for your responce.    
Two questions arise while reading the code, hope to get your directions.   
**1. why BMN use parameter 'num_sample_perbin'?**    
I can understand that in sampling mask weight whose size is N * T, we should sample once in each row, but the code may sample 'num_sample_perbin' times in each row? I didn't find any related context in the paper, hope to hear your thoughts on this parameter.
**2. Shape of BM confidence map**  
I check the paper, on ActivityNet, D and T are both set to 100, in a sense, you did the same thing(Laugh).    
Limited to my poor code capabilities，I sincerely hope to know your experimental details when you change the size of confidence map. In theory, the bigger one will get more precise result, in practice, the set of D may delete many impossible proposals.  **Did the size of BM confidence map change the results of the experiment significantly? What are their advantages and disadvantages?**      
Hope to get your responce.      
Thank you for your generosity and kindness.    
Best wishes!

**Comment by JJBOY** [comment]
Thanks for your interests.
1、real_sample_num=num_sample_perbin*num_sample. So, it's just for sampling more freatures for a proposal. 
2、The bigger map will get more precise result intuitively. Because D must less than or equal to T, so I did the experiments with D=T=200 and D=T=400. But the bigger D and T will not bring any performance improvement.

**Comment by FuNian788** [comment]
Thanks! It really helps!
